storagenode/storagenodedb/vouchers.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storagenodedb
+
+import (
+	"context"
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/storagenode/vouchers"
+)
+
+type vouchersdb struct{ *InfoDB }
+
+// Vouchers returns database for storing vouchers
+func (db *DB) Vouchers() vouchers.DB { return db.info.Vouchers() }
+
+// Vouchers returns database for storing vouchers
+func (db *InfoDB) Vouchers() vouchers.DB { return &vouchersdb{db} }
+
+// Put inserts or updates a voucher from a satellite
+func (db *vouchersdb) Put(ctx context.Context, voucher *pb.Voucher) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	id := voucher.SatelliteId
+	expiration, err := ptypes.Timestamp(voucher.GetExpiration())
+	if err != nil {
+		return ErrInfo.Wrap(err)
+	}
+
+	voucherSerialized, err := proto.Marshal(voucher)
+	if err != nil {
+		return ErrInfo.Wrap(err)
+	}
+
+	defer db.locked()()
+
+	_, err = db.db.Exec(`
+		INSERT INTO vouchers(
+			satellite_id, 
+			voucher_serialized, 
+			expiration
+		) VALUES (?, ?, ?)
+			ON CONFLICT(satellite_id) DO UPDATE SET
+				voucher_serialized = ?,
+				expiration = ?
+	`, id, voucherSerialized, expiration, voucherSerialized, expiration)
+
+	return err
+}
+
+// GetExpiring retrieves all vouchers that are expired or about to expire
+func (db *vouchersdb) GetExpiring(ctx context.Context) (satellites []storj.NodeID, err error) {
+	defer mon.Task()(&ctx)(&err)
+	defer db.locked()()
+
+	expiresBefore := time.Now().UTC().AddDate(0, 0, 3)
```

Comment:
u getting the vouchers that will be expiring in 3days from now()? I am missing something?

Reply:
That's correct. I'll leave a comment in the code to make it more clear

Reply:
In fact, I might just make it configurable